amethyst_renderer/src/pipe/pass.rs

```diff
@@ -16,8 +16,10 @@ pub trait Pass: for<'a> PassData<'a> {
     /// The pass is given an opportunity to compile shaders and store them in an `Effect`
     /// which is then passed to the pass in `apply`.
     fn compile(&mut self, effect: NewEffect) -> Result<Effect>;
+
     /// Called whenever the renderer is ready to apply the pass.  Feed commands into the
     /// encoder here.
+    // TODO: Not one of the implementations uses lifetime 'b. Is it required?
```

Comment:
What about this one?

Reply:
This one is needed, because the pass data does not have the same lifetime as the pass.

Reply:
Okay. Was just curious because all implementations of `Pass` I could discover either utilized `'a` a second time (e.g. `Pass for DrawSprite`) or the anonymous lifetime  (e.g. `Pass for DrawUi`) but never `'b` for `PassData`.